### PR TITLE
Fix "omit barline" sometimes not applying correctly

### LINKED ISSUE #22551: Barlines doesn't hide
### Type

Game behaviour

### Bug description

In this map: https://osu.ppy.sh/beatmapsets/1770207#taiko/3625974 (Antti's Oni)

Barlines at around 00:11:757 show while they are hidden in Stable

### Screenshots or videos

Skärmavbild 2023-02-08 kl  11 24 31

The barlines without hit objects on them are hidden in stable

### Version

2023.207.0

### Logs

[database.log](https://github.com/ppy/osu/files/10684405/database.log)
[network.log](https://github.com/ppy/osu/files/10684406/network.log)
[performance-draw.log](https://github.com/ppy/osu/files/10684407/performance-draw.log)
[performance-update.log](https://github.com/ppy/osu/files/10684408/performance-update.log)
[performance.log](https://github.com/ppy/osu/files/10684409/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10684411/runtime.log)

### PR DESCRIPTION
**Pull request by peppy**
This is an alternative to #22582, which increases the scope of change but reduces the complexity.

Taken from that PR:

> In stable, inherited points cannot set the omit barline effect, the checkbox is grayed out.
> Usually, inherited points would inherit the omit barline effect setting from its uninherited parent, but this is not always the case, as with [the map in the aforementioned issue](https://osu.ppy.sh/beatmapsets/1770207#taiko/3625974)
>
> It contains these two lines
> ```
> 11992,228.136882129278,4,1,0,90,1,8
> 11992,-136.979166666667,4,1,0,90,0,0
> ```

By moving the specification to the more correct location of `TimingControlPoint`, we can resolve this issue without too much extra thought.

- Closes #22551.
- Supersedes and closes #22582.

### CODE REVIEW

**Comment by peppy** [comment]
@sw1tchbl4d3r Please take a look at this and see how it looks to you. during RL discussion it was touched on that the previous PR was very complicated to understand and review. I think this approach feels better (and is a better direction in terms of exposing the setting to the user in the editor).

**Comment by sw1tchbl4d3r** [comment]
This is definitely the better approach, was worried a sudden change of a property like this might break older lazer-made maps that rely on the property being part of the effect point, but this still needs some work in the editor, in particular this:
barline

The full info "badge" gets too long to fit in the section and gets moved downwards, which results in this weird displaying of the property.

**Comment by peppy** [comment]
I've fixed the layout issue with this one, so it should be good to go.

File: osu.Game/Beatmaps/Formats/LegacyBeatmapEncoder.cs
Comment (bdach) [inline osu.Game/Beatmaps/Formats/LegacyBeatmapEncoder.cs:234]: Of note, this is a subtler-than-apparent change which makes it so that if the problematic beatmap mentioned in the issue is re-exported using lazer's editor, it will auto-fix itself, as in the problematic inherited point will have the omit bar line flag set after export. Which is neat, but also technically breaks import-export stability. But I think that's fine.

**Review by bdach — APPROVED** [review]
Definitely think this is much better (read: easier to digest) than the alternative PR.

Seeing that IRL discussions were mentioned above, I'm unsure if I'm allowed to just merge this or if I should wait for another opinion. I will say that I am relying on unit tests here and have not done much further testing on the ranked beatmap corpus (if that is something you want me to do, then let me know).

**Comment by peppy** [comment]
@bdach if you're confident it's likely fine to go ahead and merge. the RL discussion was mostly very shallow smoogi:"other PR is complex" me:"yeah but i think i have a cleaner alternative let me do that instead".

**Pull request merged**
